slider/beatmap.py

```diff
@@ -1232,6 +1232,9 @@ def __init__(self,
         self.slider_tick_rate = slider_tick_rate
         self.timing_points = timing_points
         self._hit_objects = hit_objects
+        # a (sorted) list of hitobject time's, so they can be searched with
+        # ``np.searchsorted``
+        self._hit_object_times = []
```

Comment:
We should make this a `lazyval` like other cached attributes. Then, consumers would just say `self._hit_object_times` and it would automatically be computed on the first call. For an example, see `max_combo`